### PR TITLE
Cleanup of `scyllaclient` package 

### PR DESCRIPTION
**Pull request by zimnx**
scyllaclient package was cleaned from unused code that we inherited by
copy pasting entire package from Scylla Manager.

Summary of changes:
* Autogenerated raw swagger client is now taken from github.com/scylladb/scylladb-swagger-go-client.
* Client logging is now unified with our project logging library.
* Datacenter cache was removed - we don't use Datacenter information anywhere.
* Unused functions were removed.
* Unused transport middlewares were removed.
* Unused config fields were removed.
* Added Config validation.

---

Prerequisites:
- [x] Remove replace from go.mod when `scylladb-swagger-go-client` repository transfer is finished

### CODE REVIEW

File: pkg/scyllaclient/config.go [hunk 1]
```diff
@@ -1,11 +1,11 @@
 package scyllaclient
 
 import (
+	"fmt"
 	"net/http"
 	"time"
 
-	"github.com/pkg/errors"
-	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/util/errors"
```
Comment (tnozicka) [inline pkg/scyllaclient/config.go:8]: nit: we usually  call this `apierrors` to avoid conflict with stdlib.

**Review by tnozicka — COMMENTED** [review]
lgtm, module the replace (the repo has already been transferred)

File: pkg/scyllaclient/client.go [hunk 2]
```diff
@@ -15,31 +16,41 @@ import (
...
-	scyllaOperations "github.com/scylladb/scylla-operator/pkg/scyllaclient/internal/scylla/client/operations"
 	"github.com/scylladb/scylla-operator/pkg/util/httpx"
+	scyllaclient "github.com/scylladb/scylladb-swagger-go-client/scylladb/gen/v1/client"
+	scyllaoperations "github.com/scylladb/scylladb-swagger-go-client/scylladb/gen/v1/client/operations"
 )
 
+var setOpenAPIGlobalsOnce sync.Once
+
+func init() {
+	setOpenAPIGlobalsOnce.Do(func() {
```
[the remark this answers is not in the document]
Comment (zimnx) [reply pkg/scyllaclient/client.go]: good point, removed sync.Once

File: pkg/scyllaclient/config.go (hunk 1 above)
Comment (zimnx) [reply pkg/scyllaclient/config.go]: done

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm

**Pull request merged**